src/CodeGen_LLVM.cpp

```diff
@@ -307,7 +314,12 @@ CodeGen_LLVM *CodeGen_LLVM::new_for_target(const Target &target,
         return make_codegen<CodeGen_PowerPC>(target, context);
     } else if (target.arch == Target::PNaCl) {
         return make_codegen<CodeGen_PNaCl>(target, context);
+#ifdef WITH_HEXAGON
```

Comment:
I've at least fixed the CMake build such that this should be resolved. As for the larger style question, let's defer that till later. That affects much more code than just what is in this PR.